### ISSUE TITLE
OpenCage status codes

### ISSUE THREAD

**Issue by reinososamanta** [geopy/geopy#477]
Hi, 

According to the OpenCage documentation https://opencagedata.com/api#codes, code 429 corresponds to "Too many requests (too quickly, rate limiting)" while code 402 corresponds to "Valid request but quota exceeded (payment required)". 

Geopy raises the following error when receiving code 429:

```
raise GeocoderQuotaExceeded(
                'The given key has gone over the requests limit in the 24'
                ' hour period or has submitted too many requests in too'
                ' short a period of time.'
            )
```

https://github.com/geopy/geopy/blob/313fe5c7f30e8511dedab98133a9d0cca72beaeb/geopy/geocoders/opencage.py#L214

However, code 402 is not handled, the one linked to "The given key has gone over the requests limit in the 24 hour period", which should actually correspond to exception `GeocoderQuotaExceeded`, while code 429 should be associated with another type of exception, since it should be possible to retry after some backoff time.

Is this something that could be handled? :)
Thank you!

**Comment by KostyaEsmukov** [comment]
It is a fair note that 402 should be handled differently, and I agree with that.

However, the two cases "no money on account" and "too many requests for a period of time" are currently not separated in geopy, and both are expected to be raised as `GeocoderQuotaExceeded`.

I see how that difference might be useful, so I guess we could introduce a new `GeocoderRateLimited` error, which would extend the `GeocoderQuotaExceeded` class (for backwards compatibility).

Would that be acceptable for you?

**Comment by reinososamanta** [comment]
Hi,

That would be perfect :) 

Thanks! :)

**Comment by KostyaEsmukov** [comment]
Fixed in #479, will be released as geopy 2.2.

402 should now be raised as a `GeocoderQuotaExceeded` error, and 429 as a new `GeocoderRateLimited` one.
